Yoast/javascript
# Add the title field to the sidebar

## Summary


This PR can be summarized in the following changelog entry:

* [@yoast/schema-blocks] Removes the Job Title block in favor of an input field in the Job Posting sidebar.
* [@yoast/schema-blocks] Changes `'blocks'` to `'information'` in the sidebar.

## Relevant technical choices:

* I moved the QuestionMarkLink to a separate file and made it reusable by passing the URL as a prop.
* The SidebarHeader component also moved to a separate file, and is now used in `BlockDefinition` rather than `InnerBlocksSidebar`, so that it can be shown at the top of the sidebar.
* We no longer watch the post title, since we're no longer comparing the job title to it.

## Test instructions

This PR can be tested by following these steps:

* Test this PR together with https://github.com/Yoast/wordpress-seo-premium/pull/3290.
* Create a new post and add the Job Posting block.
* See that the Job Title block is gone.
* See that the sidebar now has an input field for the Job title.
* See that the analysis has a red bullet with the text "Job title has been left empty.".
* Fill in the other required blocks (description and location), and see that the analysis conclusion stays red as long as the job title hasn't been filled in.
* See that no Schema is generated on the front-end.
* Add white space to the job title field, and see that the analysis stays red.
* Add other content to the job title field (letters and/or numbers) and see that the warning about the job title disappears, and that the analysis turns green.
* See that Schema is generated on the front-end, and that the value for `title` corresponds to what you entered in the input field.
* See that the Job title label is accompanied by a clickable help icon, which is grey but turns blue when you hoover over it (compare the designs).

**Note**
* This PR contains a bug where you can only update the Job title field if the parent block is selected. However, because we are removing the parent block in https://yoast.atlassian.net/browse/P2-940 I'm not fixing this.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes P2-866.
